crates/runtime/src/instance/allocator/pooling.rs

```diff
@@ -532,6 +571,47 @@ impl InstancePool {
         Ok(())
     }
 
+    #[cfg(feature = "memfd-allocator")]
+    fn map_instance_memfd_memories(
```

Comment:
If possible this is another place that I think would be best to share with the preexisting `set_instance_memories` method. It seems like this wants to share a fair bit of the logic and only tweaks things by acquiring a memfd slot, and that seems like it could be done in a more surgical #[cfg]'d fashion instead of duplicating the whole method.

Reply:
I unified this with the above, given the conditional-at-the-lowest-level strategy you outlined -- thanks a bunch for that, it's much cleaner now!